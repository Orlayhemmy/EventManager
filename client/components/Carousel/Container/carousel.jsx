import React from 'react';

const Carousel = (centerState) => {
  let centerList;
  let carouselIndicator;
  const { centers } = centerState;
  if (centers.length > 0) {
    centerList = centers.map((center, index) => {
      const { centerName, location, imageUrl } = center;
      return (
        <div
          className={index === 0 ? 'carousel-item active' : 'carousel-item'}
          key={index} // eslint-disable-line
        >
          <img className="d-block w-100 h-350" src={imageUrl} alt={centerName} />
          <div className="carousel-caption d-none d-md-block">
            <h5>
              {centerName}
            </h5>
            <p>
              {location}
            </p>
          </div>
        </div>
      );
    });
    carouselIndicator = centers.map((center, index) => (
      <li
        data-target="#carouselExampleIndicators"
        data-slide-to={index}
        className={index === 0 ? 'active' : ''}
        key={index} // eslint-disable-line
      />
    ));
  }
  return (
    <div className="col-lg-6 form">
      <h1>
        Our Centers
      </h1>
      <div
        id="carouselExampleIndicators"
        className="carousel slide"
        data-ride="carousel"
      >
        <ol className="carousel-indicators">
          {carouselIndicator}
        </ol>
        <div className="carousel-inner">
          {centerList}
        </div>
        <a
          className="carousel-control-prev"
          href="#carouselExampleIndicators"
          role="button"
          data-slide="prev"
        >
          <span className="carousel-control-prev-icon" aria-hidden="true" />
          <span className="sr-only">
          Previous
          </span>
        </a>
        <a
          className="carousel-control-next"
          href="#carouselExampleIndicators"
          role="button"
          data-slide="next"
        >
          <span className="carousel-control-next-icon" aria-hidden="true" />
          <span className="sr-only">
            Next
          </span>
        </a>
      </div>
    </div>
  );
};
export default Carousel;
