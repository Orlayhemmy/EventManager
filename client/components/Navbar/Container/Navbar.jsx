import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { logout } from '../../../actions/userActions';
import GuestNavbar from '../Template/Content/GuestNavbar';
import UserNavbar from '../Template/Content/Index';

/**
 * @description NavBar component
 */
export class NavBar extends React.Component {
  /**
   * @memberof NavBar
   * @method render
   * @description it renders the component
   * @returns the HTML of NavBar
   */
  render() {
    const { isAuth } = this.props.auth;
      return (
      <nav
        className="navbar navbar-expand-lg navbar-dark fixed-top"
        id="mainNav"
      >
        <Link to="/" className="navbar-brand">
          <h1>
            <span className="text-lowercase fw">e</span>Center
          </h1>
        </Link>
        <button
          className="navbar-toggler navbar-toggler-right"
          type="button"
          data-toggle="collapse"
          data-target="#navbarResponsive"
          aria-controls="navbarResponsive"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse" id="navbarResponsive">
          {isAuth ? 
            <UserNavbar navbarProps={this.props} /> :
            <GuestNavbar navbarProps={this.props} />}
        </div>
      </nav>
    );
  }
}

NavBar.propTypes = {
  auth: PropTypes.object.isRequired,
  logout: PropTypes.func.isRequired
};

function mapStateToProps(state) {
  return {
    auth: state.auth
  };
}

export default connect(mapStateToProps, { logout })(NavBar);
