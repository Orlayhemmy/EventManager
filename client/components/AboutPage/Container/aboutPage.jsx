import React from 'react';
import Navbar from '../../Navbar/Index.jsx';
import Content from '../Template/aboutPageContent';
import Footer from '../../Footer/Index';

export default class AboutPage extends React.Component {
  render() {
    const { pathname } = this.props.location;
    return (
      <div id="about-page">
        <Navbar path={pathname} />
        <Content />
        <Footer />
      </div>
    );
  }
}
