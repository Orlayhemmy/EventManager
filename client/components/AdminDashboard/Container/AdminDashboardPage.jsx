import React from 'react';
import AdminContent from '../Template/Content/AdminContent';
import Navbar from '../../Navbar/Container/Navbar';
import Footer from '../../Footer/Footer';
/**
 * @memberof AdminDashboarPage
 * @method render
 */
const AdminDashboardPage = (props) => {
  const { location: { pathname } } = props;
  return (
    <div id="center-page">
      <Navbar path={pathname} />
      <AdminContent path={pathname} />
      <Footer />
    </div>
  );
};
export default AdminDashboardPage;
