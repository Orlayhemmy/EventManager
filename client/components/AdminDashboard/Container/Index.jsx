import React from 'react';
import AdminContent from '../Template/Content/AdminContent';
import Navbar from '../../Navbar/Container/navbar';
import Footer from '../../Footer/footer';
/**
 * @memberof AdminDashMethod
 * @method render
 */
export default class AdminDashMethod extends React.Component {
  render() {
    const { pathname } = this.props.location;
    return (
      <div id="center-page">
        <Navbar path={pathname} />
        <AdminContent pathname={pathname} />
        <Footer />
      </div>
    );
  }
}
