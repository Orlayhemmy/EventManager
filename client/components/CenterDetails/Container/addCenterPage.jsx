import React from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import swal from 'sweetalert2';
import CenterForm from '../Template/Form/addCenterForm';
import Navbar from '../../Navbar/Container/navbar';
import Footer from '../../Footer/footer';

/**
 * @description AddCenterPage component
 */
export class AddCenterPage extends React.Component {
  render() {
    const { user: { isAdmin }, isAuth } = this.props.userState;
    // Check if user is logged in and is also an Admin
    if (!isAuth) {
      return <Redirect to="/" />;
    } else if (!isAdmin) {
      return <Redirect to="/dashboard" />;
    }
    const { status, message } = this.props.center;
    const { pathname } = this.props.location;

    if (status === 201) {
      swal(message);
      return <Redirect to="/admin-centers" />;
    }
    return (
      <div id="add-center">
        <Navbar path={pathname} />
        <div class="container">
          <div class="row">
            <div className="col-lg-6">
              <div class="form-outer text-center">
                <div class="form-inner">
                  <span class="logo">
                    <strong class="text-primary">add a new center</strong>
                  </span>
                  <hr />
                  <CenterForm path={pathname} />
                </div>
              </div>
            </div>
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}

const propTypes = {
  userState: PropTypes.object.isRequired,
  center: PropTypes.object.isRequired
};
const mapStateToProps = state => ({
  userState: state.auth,
  center: state.center
});
AddCenterPage.propTypes = propTypes;

export default connect(
  mapStateToProps,
  {}
)(AddCenterPage);
