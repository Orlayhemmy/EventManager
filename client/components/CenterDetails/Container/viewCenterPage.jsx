import React from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import swal from 'sweetalert2';
import PropTypes from 'prop-types';
import Navbar from '../../Navbar/Container/navbar';
import Content from '../Template/Content/centerDetailsContent';
import Footer from '../../Footer/footer';
import { logout } from '../../../actions/userActions';

/**
 * @description ViewCenterDetails component
 */
export class ViewCenterDetails extends React.Component {
  /**
   * @memberof ViewCenterDetails
   * @method logout
   * @description it calls a logout action
   * @param {object} event
   * @returns {void}
   */
  logout(e) {
    this.props.logout();
  }
  /**
   * @memberof ViewCenterDetails
   * @method render
   * @description it renders the component
   * @returns the HTML of ViewCenterDetails
   */
  render() {
    const {
      center: { status }
    } = this.props;
    //Check if user is logged in and is also an Admin
    if (!this.props.auth.isAuth) {
      return <Redirect to="/" />;
    } else if (!this.props.auth.user.isAdmin) {
      return <Redirect to="/dashboard" />;
    }
    if (status === (401 || 498)) {
      this.logout();
    } else if (status === 202) {
      swal('Center updated successfully');
    }
    const { pathname } = this.props.location;
    return (
      <div className="page-wrapper">
        <Navbar path={pathname} />
        <Content path={pathname} />
        <Footer />
      </div>
    );
  }
}
const propTypes = {
  logout: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired,
  center: PropTypes.object.isRequired
};
const mapStateToProps = state => ({
  auth: state.auth,
  center: state.center
});
ViewCenterDetails.propTypes = propTypes;

export default connect(
  mapStateToProps,
  {
    logout
  }
)(ViewCenterDetails);
