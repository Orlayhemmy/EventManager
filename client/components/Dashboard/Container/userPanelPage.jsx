/* eslint disable */
import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Redirect, Link } from 'react-router-dom';
import swal from 'sweetalert';
import { getEvents, setCurrentEvent } from '../../../actions/eventActions';
import EventForm from '../../EventPage/Template/Form/editEventForm';
import Navbar from '../../Navbar/Container/navbar';
import Footer from '../../Footer/footer';
import DeleteModal from '../../Modal/Container/deleteModal';
import { clearCenterStorage } from '../../../actions/centerActions';
import Modal from '../../Flash/modal';
import { logout } from '../../../actions/userActions';
import { getActivity } from '../../../actions/activityActions';
import DashboardContent from '../Template/Content/UserEvents';
import DashboardNotifications from '../Template/Content/UserNotifications';

/**
 * @description Dashboard component
 */
export class Dashboard extends React.Component {
  constructor() {
    super();
    this.state = {
      counter: 0
    };
    this.onClick = this.onClick.bind(this);
    this.onDelete = this.onDelete.bind(this);
    this.nextEvents = this.nextEvents.bind(this);
  }
  /**
   * @memberof Dashboard
   * @method componentWillMount
   * @description it calls an action
   * @param {void}
   * @returns {void}
   */
  componentWillMount() {
    this.props.clearCenterStorage();
    this.props.getEvents(0);
    this.props.getActivity(this.props.auth.user.id);
  }
  /**
   * @memberof Dashboard
   * @method onClick
   * @description it calls an action
   * @param {object} event
   * @returns {void}
   */
  onClick(e) {
    this.props.setCurrentEvent(e.target.id, e.target.parentNode.id);
  }
  /**
   * @memberof Dashboard
   * @method componentDidUpdate
   * @description it checks some conditions when component updates
   * @returns {void}
   */
  componentDidUpdate() {
    if (this.props.userEvent.status === 200) {
      $(document).ready(function() {
        $('#deleteModal').modal('hide');
      });
      swal(this.props.userEvent.message);
    }
  }
  /**
   * @memberof Dashboard
   * @method onDelete
   * @description it calls an action
   * @param {object} event
   * @returns {void}
   */
  onDelete(e) {
    const eventData = {
      eventId: e.target.id,
      eventName: e.target.parentNode.id
    };
    this.props.eventSelected(eventData);
  }
  /**
   * @memberof Dashboard
   * @method showDiv
   * @description it toggles div display
   * @param {object} event
   */
  showHiddenDiv(e) {
    const targetDiv = e.target.id;
    const div = document.getElementById(targetDiv);
    div.hidden = !div.hidden;
  }
  /**
   * @memberof Dashboard
   * @method logout
   * @description it calls a logout action
   * @param {object} event
   * @returns {void}
   */
  logout(e) {
    this.props.logout();
  }
  /**
   * @memberof Dashboard
   * @method nextEvents
   * @description it fetches the next centers
   * @returns {void}
   */
  nextEvents(e) {
    if (e.target.id === 'next') {
      this.setState({
        counter: this.state.counter + 1
      });
      this.props.getEvents(++this.state.counter);
    } else {
      this.setState({
        counter: this.state.counter - 1
      });
      this.props.getEvents(--this.state.counter);
    }
  }
  /**
   * @memberof Dashboard
   * @method render
   * @description it renders the component
   * @returns the HTML of Dashboard
   */
  render() {
    const { status } = this.props.userEvent;
    if (!this.props.auth.isAuth) {
      return <Redirect to="/" />;
    }
    if (status === 498 || status === 403) {
      this.logout();
    }
    const { pathname } = this.props.location;
    const { activities } = this.props.activity;
    const { message } = this.props.userEvent;
    return (
      <div id="dashboard">
        <Navbar />
        <div className="container">
          <div className="row pt-4">
            <div className="col-lg-9 col9-bg">
              <div className="row event">
                <h1>Dashboard</h1>
                <h3 className="main-color">List of events booked</h3>
                <hr />
              </div>
              <div className="row">
                <DashboardContent
                  userEvent={this.props.userEvent}
                  counter={this.state.counter}
                  onClick={this.onClick}
                  onDelete={this.onDelete}
                  nextEvents={this.nextEvents}
                />
                <DeleteModal path={pathname} />
                <Modal message={this.props.userEvent.message} />
              </div>
            </div>
            <div className="col-lg-3 col3-bg">
              <div className="p-4 fw">
                <h2>Notifications</h2>
              </div>
              <DashboardNotifications
                activities={activities}
                onClick={this.onClick}
              />
            </div>
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}
const propTypes = {
  auth: PropTypes.object.isRequired,
  userEvent: PropTypes.object.isRequired,
  activity: PropTypes.object.isRequired,
  getEvents: PropTypes.func.isRequired,
  clearCenterStorage: PropTypes.func.isRequired,
  setCurrentEvent: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
  getActivity: PropTypes.func.isRequired
};
const mapStateToProps = state => ({
  auth: state.auth,
  userEvent: state.event,
  activity: state.adminActivity
});
Dashboard.propTypes = propTypes;

export default connect(
  mapStateToProps,
  {
    setCurrentEvent,
    logout,
    getEvents,
    getActivity,
    clearCenterStorage
  }
)(Dashboard);
