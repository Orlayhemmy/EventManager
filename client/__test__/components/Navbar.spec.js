import React from 'react';
import { NavBar } from '../../components/Navbar/Container/Navbar';

const auth = {
  isAuth: true
};
describe('guest navbar', () => {
  const wrapper = shallow(<NavBar auth={auth} />);
  it('should render the component based', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
