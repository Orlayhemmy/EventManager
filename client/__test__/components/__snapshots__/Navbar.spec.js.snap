// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`guest navbar should render the component based 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar
    auth={
      Object {
        "isAuth": true,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          className="navbar-brand"
          replace={false}
          to="/"
        >
          <h1>
            <span
              className="text-lowercase fw"
            >
              e
            </span>
            Center
          </h1>
        </Link>,
        <button
          aria-controls="navbarResponsive"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler navbar-toggler-right"
          data-target="#navbarResponsive"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>,
        <div
          className="collapse navbar-collapse"
          id="navbarResponsive"
        >
          <DefaultNavbar
            navbarProps={
              Object {
                "auth": Object {
                  "isAuth": true,
                },
              }
            }
          />
        </div>,
      ],
      "className": "navbar navbar-expand-lg navbar-dark fixed-top",
      "id": "mainNav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <h1>
            <span
              className="text-lowercase fw"
            >
              e
            </span>
            Center
          </h1>,
          "className": "navbar-brand",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="text-lowercase fw"
              >
                e
              </span>,
              "Center",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "e",
                "className": "text-lowercase fw",
              },
              "ref": null,
              "rendered": "e",
              "type": "span",
            },
            "Center",
          ],
          "type": "h1",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-controls": "navbarResponsive",
          "aria-expanded": "false",
          "aria-label": "Toggle navigation",
          "children": <span
            className="navbar-toggler-icon"
          />,
          "className": "navbar-toggler navbar-toggler-right",
          "data-target": "#navbarResponsive",
          "data-toggle": "collapse",
          "type": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "navbar-toggler-icon",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <DefaultNavbar
            navbarProps={
              Object {
                "auth": Object {
                  "isAuth": true,
                },
              }
            }
          />,
          "className": "collapse navbar-collapse",
          "id": "navbarResponsive",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "navbarProps": Object {
              "auth": Object {
                "isAuth": true,
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="navbar-brand"
            replace={false}
            to="/"
          >
            <h1>
              <span
                className="text-lowercase fw"
              >
                e
              </span>
              Center
            </h1>
          </Link>,
          <button
            aria-controls="navbarResponsive"
            aria-expanded="false"
            aria-label="Toggle navigation"
            className="navbar-toggler navbar-toggler-right"
            data-target="#navbarResponsive"
            data-toggle="collapse"
            type="button"
          >
            <span
              className="navbar-toggler-icon"
            />
          </button>,
          <div
            className="collapse navbar-collapse"
            id="navbarResponsive"
          >
            <DefaultNavbar
              navbarProps={
                Object {
                  "auth": Object {
                    "isAuth": true,
                  },
                }
              }
            />
          </div>,
        ],
        "className": "navbar navbar-expand-lg navbar-dark fixed-top",
        "id": "mainNav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <h1>
              <span
                className="text-lowercase fw"
              >
                e
              </span>
              Center
            </h1>,
            "className": "navbar-brand",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="text-lowercase fw"
                >
                  e
                </span>,
                "Center",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "e",
                  "className": "text-lowercase fw",
                },
                "ref": null,
                "rendered": "e",
                "type": "span",
              },
              "Center",
            ],
            "type": "h1",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-controls": "navbarResponsive",
            "aria-expanded": "false",
            "aria-label": "Toggle navigation",
            "children": <span
              className="navbar-toggler-icon"
            />,
            "className": "navbar-toggler navbar-toggler-right",
            "data-target": "#navbarResponsive",
            "data-toggle": "collapse",
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "navbar-toggler-icon",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <DefaultNavbar
              navbarProps={
                Object {
                  "auth": Object {
                    "isAuth": true,
                  },
                }
              }
            />,
            "className": "collapse navbar-collapse",
            "id": "navbarResponsive",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "navbarProps": Object {
                "auth": Object {
                  "isAuth": true,
                },
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
